### ISSUE TITLE
Bug for VizdoomEnv

### ISSUE THREAD

**Issue by caozhenxiang-kouji** [Farama-Foundation/ViZDoom#574]
There seems to be a serious bug with your VizdoomEnv.

When I use the following code to build a doom environment:
```python
import gymnasium
from vizdoom import gymnasium_wrapper

env = gymnasium.make("VizdoomHealthGathering-v0", frame_skip=4)
```
And I try to use the following code to step forward for the environment:
```python
obs, reward, terminated, truncated, info = env.step(env.action_space.sample())
```
And my agent is always moving forward and never turn left or right.

It seems that if the ```env.step``` function takes in an int type, the ```env_action``` in ```env.step``` function can produce correct results like [0,0,0],[0,0,1],[0,1,0],[1,0,0]. However when the ```env.step``` function takes in a numpy.int64 type, the ```env_action``` will turn into [0,0,0],[1,1,1],[2,2,2],[3,3,3], which forces the agent to keep moving forward.

My way of solving this bug is to write another gym.Wrapper to overwrite the step function:
```python
class ChangeAction(gym.Wrapper):
    def step(self, action):
        action = int(action)
        return self.env.step(action)
```
It doesn't seem to be an elegant way since every time you make a environment you should wrap it again. But it works fine.

Hopes you can fix this bug soon!

**Comment by mwydmuch** [comment]
Hi @caozhenxiang-kouji, thank you very much for the report; this is indeed a serious bug! The fix will be merged soon, and a new patch release should be available on PyPI in a few hours.

**Comment by mwydmuch** [comment]
I released new version 1.2.3 that fixes the problem, it should be deployed to PyPI in ~2-3h. Once again, thanks for reporting the problem.

**Comment by caozhenxiang-kouji** [comment]
Great, thanks
